AutoFixture.AutoFakeItEasy - Latest FakeItEasy Support

It appears that the latest AutoFixture.AutoFakeItEasy is not able to work with the latest FakeItEasy version. Are there plans to support the latest FakeItEasy?

`warning NU1608: Detected package version outside of dependency constraint: AutoFixture.AutoFakeItEasy 4.14.0 requires FakeItEasy (>= 4.9.0 && < 6.0.0) but version FakeItEasy 6.2.1 was resolved.`

--- a/Src/All.sln
+++ b/Src/All.sln
@@ -88,6 +88,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AutoFakeItEasy.FakeItEasy5U
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AutoNSubstitute.NSubstitute4UnitTest", "AutoNSubstitute.NSubstitute4UnitTest\AutoNSubstitute.NSubstitute4UnitTest.csproj", "{4BDBE8A1-CF75-4049-9B3D-B194B1F45369}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "AutoFakeItEasy.FakeItEasy6UnitTest", "AutoFakeItEasy.FakeItEasy6UnitTest\AutoFakeItEasy.FakeItEasy6UnitTest.csproj", "{311A3010-B938-4832-9D3A-21A1A56FE5EA}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
@@ -317,6 +319,12 @@ Global
 		{4BDBE8A1-CF75-4049-9B3D-B194B1F45369}.Release|Any CPU.Build.0 = Release|Any CPU
 		{4BDBE8A1-CF75-4049-9B3D-B194B1F45369}.Verify|Any CPU.ActiveCfg = Verify|Any CPU
 		{4BDBE8A1-CF75-4049-9B3D-B194B1F45369}.Verify|Any CPU.Build.0 = Verify|Any CPU
+		{311A3010-B938-4832-9D3A-21A1A56FE5EA}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{311A3010-B938-4832-9D3A-21A1A56FE5EA}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{311A3010-B938-4832-9D3A-21A1A56FE5EA}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{311A3010-B938-4832-9D3A-21A1A56FE5EA}.Release|Any CPU.Build.0 = Release|Any CPU
+		{311A3010-B938-4832-9D3A-21A1A56FE5EA}.Verify|Any CPU.ActiveCfg = Verify|Any CPU
+		{311A3010-B938-4832-9D3A-21A1A56FE5EA}.Verify|Any CPU.Build.0 = Verify|Any CPU
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE

--- a/Src/AutoFakeItEasy.FakeItEasy6UnitTest/AutoFakeItEasy.FakeItEasy6UnitTest.csproj
+++ b/Src/AutoFakeItEasy.FakeItEasy6UnitTest/AutoFakeItEasy.FakeItEasy6UnitTest.csproj
@@ -1,0 +1,33 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\Common.props" />
+  <Import Project="..\Common.Test.props" />
+  <Import Project="..\Common.Test.xUnit.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netcoreapp1.1;netcoreapp2.1</TargetFrameworks>
+    <AssemblyTitle>AutoFakeItEasy.FakeItEasy5.UnitTest</AssemblyTitle>
+    <AssemblyName>AutoFixture.AutoFakeItEasy.FakeItEasy5UnitTest</AssemblyName>
+    <RootNamespace>AutoFixture.AutoFakeItEasy.UnitTest</RootNamespace>
+
+    <DefineConstants>$(DefineConstants);CAN_FAKE_DELEGATES;HAS_A_CALL_TO_SET_SPECIFIER</DefineConstants>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="FakeItEasy" Version="[6.2.1]" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Include="..\AutoFakeItEasyUnitTest\**\*.cs" Link="%(RecursiveDir)%(Filename)%(Extension)" />
+    <Compile Remove="..\AutoFakeItEasyUnitTest\obj\**" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\AutoFakeItEasy\AutoFakeItEasy.csproj" />
+    <ProjectReference Include="..\AutoFixture\AutoFixture.csproj" />
+    <ProjectReference Include="..\TestTypeFoundation\TestTypeFoundation.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+</Project>

--- a/Src/AutoFakeItEasy/AutoFakeItEasy.csproj
+++ b/Src/AutoFakeItEasy/AutoFakeItEasy.csproj
@@ -20,9 +20,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="FakeItEasy" Version="[1.7.4109.1,6.0.0)" Condition=" '$(TargetFramework)'=='net452' " />
-    <PackageReference Include="FakeItEasy" Version="[3.0.0,6.0.0)" Condition=" '$(TargetFramework)'=='netstandard1.6' " />
-    <PackageReference Include="FakeItEasy" Version="[4.9.0,6.0.0)" Condition=" '$(TargetFramework)'=='netstandard2.0' " />
+    <PackageReference Include="FakeItEasy" Version="[1.7.4109.1,7.0.0)" Condition=" '$(TargetFramework)'=='net452' " />
+    <PackageReference Include="FakeItEasy" Version="[3.0.0,7.0.0)" Condition=" '$(TargetFramework)'=='netstandard1.6' " />
+    <PackageReference Include="FakeItEasy" Version="[4.9.0,7.0.0)" Condition=" '$(TargetFramework)'=='netstandard2.0' " />
   </ItemGroup>
 
   <ItemGroup>

--- a/Src/AutoFakeItEasy/FakeObjectCall.cs
+++ b/Src/AutoFakeItEasy/FakeObjectCall.cs
@@ -18,6 +18,7 @@ namespace AutoFixture.AutoFakeItEasy
         private const string SetReturnValueMethodName = "SetReturnValue";
         private const string SetArgumentValueMethodName = "SetArgumentValue";
         private const string ArgumentsPropertyName = "Arguments";
+        private const string ReturnValuePropertyName = "ReturnValue";
 
         private readonly IFakeObjectCall wrappedCall;
 
@@ -32,11 +33,17 @@ namespace AutoFixture.AutoFakeItEasy
 
         public void SetReturnValue(object value)
         {
-            this.InvokeWrappedMethod(SetReturnValueMethodName, value);
+            // This invocation is valid for FakeItEasy prior to v6.
+            if (this.TryInvokeWrappedMethod(SetReturnValueMethodName, value)) return;
+
+            // This invocation will be used for FakeItEasy v6 and later.
+            if (this.TryInvokeWrappedPropertySetter(ReturnValuePropertyName, value)) return;
+
+            throw new MissingMethodException("Unable to find a supported return value setter member");
         }
 
         public void SetArgumentValue(int index, object value) =>
-            this.InvokeWrappedMethod(SetArgumentValueMethodName, index, value);
+            this.TryInvokeWrappedMethod(SetArgumentValueMethodName, index, value);
 
         private object InvokeWrappedPropertyGetter(string propertyName)
         {
@@ -51,17 +58,24 @@ namespace AutoFixture.AutoFakeItEasy
             return propertyInfo.GetValue(this.wrappedCall);
         }
 
-        private void InvokeWrappedMethod(string methodName, params object[] parameters)
+        private bool TryInvokeWrappedPropertySetter(string propertyName, object value)
+        {
+            var callType = this.wrappedCall.GetType();
+            var propertyInfo = callType.GetProperty(propertyName);
+            if (propertyInfo == null) return false;
+
+            propertyInfo.SetValue(this.wrappedCall, value);
+            return true;
+        }
+
+        private bool TryInvokeWrappedMethod(string methodName, params object[] parameters)
         {
             var callType = this.wrappedCall.GetType();
             var methodInfo = callType.GetMethod(methodName);
-            if (methodInfo == null)
-            {
-                throw new MissingMethodException(string.Format(CultureInfo.CurrentCulture,
-                    "Method {0} cannot be found on {1}", methodName, callType.FullName));
-            }
+            if (methodInfo == null) return false;
 
             methodInfo.Invoke(this.wrappedCall, parameters);
+            return true;
         }
     }
 }

